pkg/asset/cluster/aws/aws.go

```diff
@@ -19,3 +25,47 @@ func Metadata(clusterID, infraID string, config *types.InstallConfig) *aws.Metad
 		}},
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID string, platform *awstypes.Platform) error {
+	if len(platform.Subnets) == 0 {
+		return nil
+	}
+
+	metas, err := awsconfig.SubnetMetadata(ctx, platform.Subnets)
+	if err != nil {
+		return err
+	}
+
+	arns := make([]string, 0, len(metas)+1)
+	for _, meta := range metas {
+		arns = append(arns, meta.ARN)
+	}
+
+	session, err := awsconfig.GetSession()
```

Comment:
can we get this client passed to this function rather than create a session here.

Reply:
Done since the reroll to build on #2512.